core/src/HIP/Kokkos_HIP_ParallelScan_Range.hpp

```diff
@@ -169,7 +169,7 @@ class ParallelScanHIPBase {
       }
 
       // Make sure the write is seen by all threads
-      __threadfence_block();
+      __syncthreads();
```

Comment:
Cuda has `__syncwarp(MASK);` here.

Reply:
> You need `__syncwarp` in CUDA because of the Independent Thread Scheduling. For AMD, all the threads are executed in lock steps, so there is no need to wait for all the threads in the warp. I am not convince this change is necessary.

If none of AMD GPUs supports independent thread scheduling, yes, we don't need to change this.
(It seems that MI300A does not support the independent thread scheduling - the test succeeds without this change.)